### PR TITLE
feat(frost): Add signature to signing commitment

### PR DESCRIPTION
**Pull request by rohanjadvani**

### CODE REVIEW

File: src/signing_commitment.rs [hunk 1]
```diff
@@ -6,32 +6,79 @@ use std::io;
...
+        &identity.serialize()[..],
+        &hiding.serialize(),
+        &binding.serialize(),
+    ];
+    let mut slice = &mut data[..];
+    for part in parts {
+        slice[..part.len()].copy_from_slice(part);
+        slice = &mut slice[part.len()..];
+    }
+    assert_eq!(slice.len(), 0);
```
Comment (andiflabs) [inline src/signing_commitment.rs]: Nit: `debug_assert_eq`

File: src/signing_commitment.rs [hunk 2]
```diff
@@ -6,32 +6,79 @@ use std::io;
...
             identity,
             hiding,
             binding,
+            signature,
+        };
+
+        signing_commitment.verify().map(|_| signing_commitment)
+    }
+
+    pub fn from_frost(
```
Comment (andiflabs) [inline src/signing_commitment.rs:61]: Nit: `#[must_use]`

**Review by andiflabs — APPROVED** [review]

File: src/signing_commitment.rs (hunk 1 above)
Comment (rohanjadvani) [reply src/signing_commitment.rs]: Updated, but do we need to update [this](https://github.com/iron-fish/ironfish-frost/blob/main/src/participant.rs#L56) too?

Comment (andiflabs) [reply src/signing_commitment.rs]: Yeah, that'd be nice

**Pull request merged**